### PR TITLE
Add test for limiting amount of computed paths #4877

### PR DESCRIPTION
**Pull request by vkhalin**
Test for #4877 and fix for NB service test implementation

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy [hunk 1]
```diff
@@ -45,6 +46,13 @@ class PathsSpec extends HealthCheckSpecification {
             }
         }
 
+        then: "Maximum count of paths can be changed during PCE calculations"
+        def limited_paths = northbound.getPaths(switchPair.src.dpId, switchPair.dst.dpId,
+                null, null, null, null, 1)
+        assert limited_paths.paths.size() == 1
+        assert paths.paths.size() > limited_paths.paths.size()
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy:53]: L26 should be improved.
you need to be sure that `switchPair` has two paths at least.

**Review by andriidovhan — APPROVED** [review]

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy (hunk 1 above)
Comment (vkhalin) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy:53]: I do not think that we need exactly what you asking for in this particular case. We need to check that max_path_count var can limit response output to some number of available paths. In our case all switch pairs will have more than 1 path, so our asserts will be always correct

Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy:53]: agree, I forgot implementation of `getNotNeighboringSwitchPair`. 
It returns switchPair with two paths at least.

Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy:53]: except that it is not)

Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathsSpec.groovy:53]: it returns a switchpair with the shortest path being of size >2 (means that switches are not neighbours, just as method name implies)

**Review by niksv — APPROVED** [review]

**Pull request merged**